### PR TITLE
Overhaul module reference page

### PR DESCRIPTION
**Pull request by s-makin**
```
Module reference page was previously long lists of descriptions that were hard to consume.
- Add tabs to split up sections within module reference
- Tab syncing is an option, but may not be helpful here so left it off by default
- Subheadings can't be used within tabs, so changed headers in cc_ssh module description
```

### CODE REVIEW

**Comment by igalic** [comment]
is there a way to preview this?

File: cloudinit/config/schema.py [hunk 1]
```diff
@@ -170,11 +182,11 @@ def handle_problems(problems, prefix):
 
         self.schema_errors = handle_problems(
             schema_errors,
-            prefix="Cloud config schema errors: ",
+            prefix=f"Cloud config schema errors: ",
         )
         self.schema_deprecations = handle_problems(
             schema_deprecations,
-            prefix="Cloud config schema deprecations: ",
+            prefix=f"Cloud config schema deprecations: ",
```
Comment (TheRealFalcon) [inline cloudinit/config/schema.py:189]: f-string isn't needed here

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Fancy! There's a few small inline issues, but otherwise LGTM.

Tests need to be updated. I'm assuming you want some help with that?

**Comment by s-makin** [comment]
> Tests need to be updated. I'm assuming you want some help with that?

Thanks for the review, I'll fix the nits - I made a bunch of experimental changes in trying to solve the "indentation" errors, and definitely forgot to undo those f-strings. 

Yes please on the help with tests!

**Comment by TheRealFalcon** [comment]
@s-makin , I rebased the branch and fixed tests. I took the liberty of pushing directly to your branch. If that's a problem, I can always undo it. If any other CI fails I can take a look.

**Pull request merged**